src/utils/atomWithDefault.ts

```diff
@@ -0,0 +1,26 @@
+import { atom, PrimitiveAtom } from 'jotai'
+import type { Read } from '../core/types'
+
+export function atomWithDefault<Value>(
+  getDefault: Read<Value>
+): PrimitiveAtom<Value> {
+  const overwrittenAtom = atom(false)
+  const anAtom: PrimitiveAtom<Value> = atom(
+    (get) => {
+      if (get(overwrittenAtom)) {
+        return get(anAtom)
+      }
+      return getDefault(get)
+    },
+    (get, set, update) => {
+      set(overwrittenAtom, true)
+      set(
+        anAtom,
+        typeof update === 'function'
```

Comment:
Can it be a promise with this impl?

Reply:
I don't think so. The created atom is just like a normal primitive atom.

Reply:
Is that acceptable?

Reply:
I mean it should be the same as the normal `atom(...)`. As far as I remember, if we set atom with a promise, it works for some case, but had a limitation or pitfall. At least the type doesn't allow it, so we are not currently supporting such a pattern.